packages/react-dom/src/events/accumulateEnterLeaveListeners.js

```diff
@@ -0,0 +1,139 @@
+/**
+ * Copyright (c) Facebook, Inc. and its affiliates.
+ *
+ * This source code is licensed under the MIT license found in the
+ * LICENSE file in the root directory of this source tree.
+ *
+ * @flow
+ */
+
+import type {Fiber} from 'react-reconciler/src/ReactFiber';
+import type {ReactSyntheticEvent} from 'legacy-events/ReactSyntheticEventType';
+
+import getListener from 'legacy-events/getListener';
+import {HostComponent} from 'react-reconciler/src/ReactWorkTags';
+
+function getParent(inst: Fiber | null): Fiber | null {
+  if (inst === null) {
+    return null;
+  }
+  do {
+    inst = inst.return;
+    // TODO: If this is a HostRoot we might want to bail out.
+    // That is depending on if we want nested subtrees (layers) to bubble
+    // events to their parent. We could also go through parentNode on the
+    // host node but that wouldn't work for React Native and doesn't let us
+    // do the portal feature.
+  } while (inst && inst.tag !== HostComponent);
+  if (inst) {
+    return inst;
+  }
+  return null;
+}
+
+/**
+ * Return the lowest common ancestor of A and B, or null if they are in
+ * different trees.
+ */
+function getLowestCommonAncestor(instA: Fiber, instB: Fiber): Fiber | null {
```

Comment:
Btw you inlined these functions in the responder event plugins too. They used to be in the tree traversal module

Reply:
I altered the PR to inline all this code into their relevant places so we don't need them for ReactDOM.